### ISSUE TITLE
[1.5.10-beta02] syncComposeResourcesForIos is incompatible with Gradle Configuration Cache

### ISSUE THREAD

**Issue by chrisbanes** [JetBrains/compose-multiplatform#3744]
**Describe the bug**
Tried out turning the Gradle configuration cache again with Kotlin 1.9.20-Beta2 (since that apparently fixes most of the K/N issues), but it looks like the `syncComposeResourcesForIos` also has issues.

**Versions**
- Kotlin version* 1.9.20-Beta2
- Compose Multiplatform version*: 1.5.10-beta02
- OS version(s)* (required for Desktop and iOS issues): Mac
- OS architecture (x86 or arm64): ARM64

```
org.gradle.configurationcache.ConfigurationCacheError: Configuration cache state could not be cached: field `resourceFiles` of task `:shared:app:syncComposeResourcesForIos` of type `org.jetbrains.compose.experimental.uikit.tasks.SyncComposeResourcesForIosTask`: error writing value of type 'org.gradle.api.internal.provider.TransformBackedProvider'
	at org.gradle.configurationcache.problems.ProblemsListener$DefaultImpls.onError(ProblemsListener.kt:37)
	at org.gradle.configurationcache.problems.ConfigurationCacheProblems.onError(ConfigurationCacheProblems.kt:38)
	at org.gradle.configurationcache.serialization.AbstractIsolateContext.onError(Contexts.kt:418)
	at org.gradle.configurationcache.serialization.beans.BeanPropertyWriterKt.writeNextProperty(BeanPropertyWriter.kt:100)
	at org.gradle.configurationcache.serialization.beans.BeanPropertyWriter.writeStateOf(BeanPropertyWriter.kt:53)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec$writeTask$3$2$1.invokeSuspend(TaskNodeCodec.kt:103)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec$writeTask$3$2$1.invoke(TaskNodeCodec.kt)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec$writeTask$3$2$1.invoke(TaskNodeCodec.kt)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodecKt.withTaskOf(TaskNodeCodec.kt:237)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodecKt.access$withTaskOf(TaskNodeCodec.kt:1)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec.writeTask(TaskNodeCodec.kt:97)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec.encode(TaskNodeCodec.kt:74)
	at org.gradle.configurationcache.serialization.codecs.TaskNodeCodec.encode(TaskNodeCodec.kt:67)
```

**Comment by eymar** [comment]
Reproduced with the template https://github.com/JetBrains/compose-multiplatform-ios-android-template :

```
* What went wrong:
Configuration cache state could not be cached: field `resourceFiles` of task `:shared:syncComposeResourcesForIos` of type `org.jetbrains.compose.experimental.uikit.tasks.SyncComposeResourcesForIosTask`: error writing value of type 'org.gradle.api.internal.provider.TransformBackedProvider'
> Could not infer iOS target platform. Make sure to build via XCode (directly or via Kotlin Multiplatform Mobile plugin for Android Studio)
....
Configuration cache entry discarded due to serialization error.
```

gradle.properties:
```
kotlin.version=1.9.20-Beta2
compose.version=1.5.10-beta02
org.gradle.configuration-cache=true
```

**Comment by eymar** [comment]
The fix was merged: https://github.com/JetBrains/compose-multiplatform/pull/3764
Should be available in 1.5.10

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3744">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
